reference/promotions.v3.yml

```diff
@@ -888,6 +905,85 @@ components:
               description: Link to the next page returned in the response.
           description: Pagination links for the previous and next parts of the whole collection.
       description: 'Data about the response, including pagination and collection totals.'
+    DeprecatedPagination:
+      title: Pagination
+      type: object
+      deprecated: true
+      properties:
+        total:
+          type: integer
+          description: Total number of items in the result set.
+        count:
+          type: integer
+          description: Total number of items in the collection response.
+        per_page:
+          type: integer
+          description: 'The amount of items returned in the collection per page, controlled by the limit of items per page parameter.'
+        current_page:
+          type: integer
+          description: The page you are currently on within the collection.
+        total_pages:
+          type: integer
+          description: The total number of pages in the collection.
+        links:
+          type: object
+          properties:
+            previous:
+              type: string
+              description: Link to the previous page returned in the response.
+            current:
+              type: string
+              description: Link to the current page returned in the response.
+            next:
+              type: string
+              description: Link to the next page returned in the response.
+          description: Pagination links for the previous and next parts of the whole collection.
+      description: >
+        Data about the response, including pagination and collection totals.
+        This property has been deprecated and cursor_pagination should be used instead.
+    CursorPagination:
+      title: Cursor Pagination
+      type: object
+      required: [
+        count,
+        per_page,
+        links
+      ]
+      properties:
+        count:
+          type: integer
+          description: Total number of items in the result set.
+          example: 12
+        per_page:
+          type: integer
+          description: 'The amount of items returned in the collection per page, controlled by the limit of items per page parameter.'
+          example: 12
+        start_cursor:
+          type: string
+          description: >
+            The cursor to the first item in the result set.
+            Can be used with the "before" query parameter to paginate backwards.
+          example: 'eyJpZCI6IjIzNzU1NyJ9'
+        end_cursor:
+          type: string
+          description: >
+            The cursor to the last item in the result set.
+            Can be used with the "after" query parameter to paginate forwards.
+          example: 'eyJpZCI6IjIzNzU1NyJ9'
```

Comment:
Do we need these fields, given we already have `previous` and `next` links ?

Reply:
Using the cursors in `previous` and `next` would require the FE to parse the url query params. For some frontends it would be nicer to be able to read the cursor value directly